code/data_structures/tree/tree_comparer/test_tree_comparer.cpp

```diff
@@ -0,0 +1,102 @@
+#ifndef TEST_TREE_COMPARER
+#define TEST_TREE_COMPARER
+
+#include "catch.hpp"
+#include "tree_serializer.cpp"
+#include "tree_comparer.cpp"
+
+TEST_CASE("check two tree is same", "[isSameTree]") {
+    TreeSerializer serializer;
+    std::shared_ptr<TreeNode<int> > root_a, root_b;
+    TreeComparer<int> comparer;
+
+    SECTION("has empty tree") {
+        root_a = serializer.deserialize("# ");
+        root_b = serializer.deserialize("# ");
+        REQUIRE(comparer.isSameTree(root_a, root_b));
+
+        root_a = serializer.deserialize("1 # # ");
+        root_b = serializer.deserialize("1 # # ");
+        REQUIRE(comparer.isSameTree(root_a, root_b));
+
+        root_a = serializer.deserialize("# ");
+        root_b = serializer.deserialize("1 # # ");
+        REQUIRE_FALSE(comparer.isSameTree(root_a, root_b));
+        std::swap(root_a, root_b);
```

Comment:
Include \<algorithm> (even though unnecessary, but clearer).

Reply:
Don't forget to do this!

Reply:
I added <utility>~

Reply:
Same thing, doesn't matter.